plugins/modules/zos_volume_init.py

```diff
@@ -0,0 +1,242 @@
+    required: false
+    type: str
+  vtoc_tracks:
+    description:
+      - The number of tracks to initialize the VTOC with.
+      - The VTOC will be placed at cylinder 0 head 1 for the number of tracks specified.
+      - Typically, ICKDSF will default the size to the number of tracks in a cylinder minus 1.
+      - For a 3390, the default is cylinder 0, track 1 for 14 tracks
+    required: false
+    type: int
+  index:
+    description:
+      - Create a VTOC index during volume initialization.
+      - The index size will be based on the size of the volume and the size of the VTOC that was created.
+      - The index will be placed on the volume after the VTOC.
+      - Set to false to not generate an index.
+    required: false
+    type: bool
+    default: true
+  sms_managed:
+    description:
+      - Assigned to be managed by Storage Management System (SMS).
+    type: bool
+    required: false
+    default: true
+  verify_no_data_sets_exist:
+    description:
+      - Verify if data sets other than the VTOC index data set and/or VVDS exist on the volume to be initialized.
+      - Beware that z/OS system defaults can override ICKDSF parameters.
+    required: false
+    type: bool
+    default: true
+"""
+EXAMPLES = r"""
+- name: Initialize target volume with all default options. Target volume address is '1234', set volume name to 'DEMO01'.
+        Target volume is checked to ensure it is offline and contains no data sets. Volume is SMS managed, has an index
+        and VTOC size defined by the system.
+  zos_volume_init:
+    volume_address: "1234"
+    volid: "DEMO01"
+
+- name: Initialize target volume with all default options same as above and additionally check the existing volid
+        matches the given value 'DEMO02' before re-initializing the volume and renaming it to 'DEMO01'
+  zos_volume_init:
+    volume_address: "1234"
+    volid: "DEMO01"
+    verify_volid: "DEMO02"
+
+- name: Initialize non-SMS managed target volume with all the default options.
+  zos_volume_init:
+    volume_address: "1234"
+    volid: "DEMO01"
+    sms_managed: no
+
+- name: Initialize a new SMS managed DASD volume with new volume serial 'e8d8' with 30 track VTOC, an index, as long as
+        the existing volume serial is 'ine8d8' and there are no pre-existing data sets on the target. The check to see
+        if volume is online before intialization is skipped.
+  zos_volume_init:
+    volume_address: e8d8
+    vtoc_tracks: 30
+    index: yes
+    sms_managed: yes
+    volid: ine8d8
+    verify_existing_volid: ine8d8
+    verify_no_data_sets_exist: yes
+    verify_offline: no
+
+- name: Initialize 3 new DASD volumes (0901, 0902, 0903) for use on a z/OS system as 'DEMO01', 'DEMO02', 'DEMO03'
+        using Ansible loops.
+  zos_volume_init:
+    volume_address: "090{{ item }}"
+    volid: "DEMO0{{ item }}"
+  loop: "{{ range(1, 4, 1) }}"
+"""
+RETURN = r"""
+command:
+  description: INIT command issued to ICKDSF tool.
```

Comment:
This is the raw command that is passed to ICKDSF command "init". Do we want to remove most/all references to the underlying ickdsf utility? I think the suggested edit is inaccurate.

Reply:
I was going for masking the utilities we use, there are a few ways at looking at this, our initial intention was to shield users from the complexity of z/OS, as much a we try we can't always achieve that and terms do bubble up. 

When you don't know much about ICKDSF, not sure this means anything to any one "INIT command issued to ICKDSF tool.", not even all system folks know this utility, that was the direction I was taking. 

I think it might better to describe how we achieve creating volumes the documentation section using ICKDSF where we can be a bit more verbose over this short 1-liner, or explain it further detail here so its not so cryptic to folks.